OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/OSOutcomeEventsV1Repository.kt

```diff
@@ -0,0 +1,69 @@
+import org.json.JSONException
+
+internal class OSOutcomeEventsV1Repository(logger: OSLogger,
+                                           outcomeEventsCache: OSOutcomeEventsCache,
+                                           outcomeEventsService: OutcomeEventsService) : OSOutcomeEventsRepository(logger, outcomeEventsCache, outcomeEventsService) {
+    override fun requestMeasureOutcomeEvent(appId: String, deviceType: Int, eventParams: OSOutcomeEventParams, responseHandler: OneSignalApiResponseHandler) {
+        val event = OutcomeEvent.fromOutcomeEventParamsV2toOutcomeEventV1(eventParams)
+        when (event.session) {
+            OSInfluenceType.DIRECT -> requestMeasureDirectOutcomeEvent(appId, deviceType, event, responseHandler)
+            OSInfluenceType.INDIRECT -> requestMeasureIndirectOutcomeEvent(appId, deviceType, event, responseHandler)
+            OSInfluenceType.UNATTRIBUTED -> requestMeasureUnattributedOutcomeEvent(appId, deviceType, event, responseHandler)
+            else -> {
+            }
+        }
+    }
+
+    private fun requestMeasureDirectOutcomeEvent(appId: String, deviceType: Int, event: OutcomeEvent, responseHandler: OneSignalApiResponseHandler) {
+        try {
+            event.toJSONObjectForMeasure()
+                    .put(APP_ID, appId)
+                    .put(DEVICE_TYPE, deviceType)
+                    .put(DIRECT, true)
+                    .also { jsonObject ->
+                        outcomeEventsService.sendOutcomeEvent(jsonObject, responseHandler)
+                    }
+        } catch (e: JSONException) {
+            logger.error("Generating direct outcome:JSON Failed.", e)
+        }
+    }
+
+    private fun requestMeasureIndirectOutcomeEvent(appId: String, deviceType: Int, event: OutcomeEvent, responseHandler: OneSignalApiResponseHandler) {
+        try {
+            event.toJSONObjectForMeasure()
+                    .put(APP_ID, appId)
+                    .put(DEVICE_TYPE, deviceType)
+                    .put(DIRECT, false)
+                    .also { jsonObject ->
+                        outcomeEventsService.sendOutcomeEvent(jsonObject, responseHandler)
+                    }
+        } catch (e: JSONException) {
+            logger.error("Generating indirect outcome:JSON Failed.", e)
+        }
+    }
+
+    private fun requestMeasureUnattributedOutcomeEvent(appId: String, deviceType: Int, event: OutcomeEvent, responseHandler: OneSignalApiResponseHandler) {
+        try {
+            event.toJSONObjectForMeasure()
+                    .put(APP_ID, appId)
+                    .put(DEVICE_TYPE, deviceType)
+                    .also { jsonObject ->
+                        outcomeEventsService.sendOutcomeEvent(jsonObject, responseHandler)
+                    }
+        } catch (e: JSONException) {
+            logger.error("Generating unattributed outcome:JSON Failed.", e)
+        }
+    }
+
+    companion object {
+        private const val DIRECT = "direct"
```

Comment:
Might be better to use just object so there isn't a companion object under the hood.
https://stackoverflow.com/a/50520935
@Jeasmine Thoughts on this?

Reply:
Agree, still need to fix the companion for DB, but I need to rebase first due to some changes on that classes, I need them in order to don't do too much re-work

Reply:
👍 I see this is now cleaned up
https://github.com/OneSignal/OneSignal-Android-SDK/pull/1042/files#diff-ebb1a5f7f3ff02ae231e90aa99ed7818R3